### PR TITLE
Rework runconfig to use options on init

### PR DESCRIPTION
**Pull request by Itxaka**
On sync with chroot for example, hopefully with everything going forward so we only have to set it on struct init

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

**Review by mudler — APPROVED** [review]
Love it!

File: pkg/types/v1/config.go [hunk 1]
```diff
@@ -26,6 +28,28 @@ const (
 	BOOT = "boot"
 )
 
+type RunConfigOptions func(a *RunConfig) error
+
+func WithFs(fs afero.Fs) func(r *RunConfig) error {
```
Comment (mudler) [inline pkg/types/v1/config.go:33]: Maybe we should start to think to document (with comments) the options? :thinking:

Comment (Itxaka) [reply pkg/types/v1/config.go:33]: good point, godoc or plain old docs?

Comment (mudler) [reply pkg/types/v1/config.go:33]: godoc please :) ! :rocket:

**Pull request merged**